modules/apps/frontend-data-set/frontend-data-set-admin-api/src/main/java/com/liferay/frontend/data/set/admin/manager/FDSAdminObjectDefinitionsManager.java

```diff
@@ -0,0 +1,17 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.frontend.data.set.admin.manager;
+
+import com.liferay.portal.kernel.model.Company;
+
+/**
+ * @author Daniel Sanz
+ */
+public interface FDSAdminObjectDefinitionsManager {
```

Comment:
I tried refactoring solution to avoid using public interface:
- SF indeed complains about naming service like the class. There is a [long list of exceptions](https://github.com/liferay/liferay-portal/blob/master/source-formatter.properties#L808-L976), but it is strongly implied that this is a bad practice and the goal is to reduce this list.
- Adding internal interface is also not allowed by SF.
- Overall solution does not work with `@Component(service = {})`
- Manager could be POJO class. This is a legitimate solution, but:
  - there is no convenient optimization mechanism for checked companies, and I'd prefer to avoid static classes
  - it would require a larger refactor of data generation, as we couldn't use `@Reference`s 

So in the end, I fell back to the original solution, with just some minor renames and tweaks.